src/callframe.rs

```diff
@@ -152,6 +152,16 @@ impl Callframe {
                 .map(|f| f.previous_frame_gas)
                 .sum::<u32>()
     }
+
+    pub fn is_kernel_mode(&self) -> bool {
+        address_is_kernel(&self.address)
+    }
+}
+
+pub fn address_is_kernel(address: &H160) -> bool {
```

Comment:
You can write this as `address < (1 << 16).into()`.

Reply:
Worth checking with compiler explorer what code that produces. We could also save a is_kernel boolean for each frame.